packages/react/test/specs/lib/AutoFocusZone-test.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react'
+import * as ReactTestUtils from 'react-dom/test-utils'
+
+import { FocusZone, AutoFocusZone } from '../../../src/lib/accessibility/FocusZone'
+
+// rAF does not exist in node - let's mock it
+window.requestAnimationFrame = (callback: FrameRequestCallback) => {
+  const r = window.setTimeout(callback, 0)
+  jest.runAllTimers()
+  return r
+}
+
+const animationFrame = () => new Promise(resolve => window.requestAnimationFrame(resolve))
+jest.useFakeTimers()
+
+describe('AutoFocusZone', () => {
+  let lastFocusedElement: HTMLElement | undefined
+
+  const _onFocus = (ev: any): void => (lastFocusedElement = ev.target)
+
+  const setupElement = (
+    element: HTMLElement,
+    {
+      clientRect,
+      isVisible = true,
+    }: {
+      clientRect: {
+        top: number
+        left: number
+        bottom: number
+        right: number
+      }
+      isVisible?: boolean
+    },
+  ): void => {
+    element.getBoundingClientRect = () => ({
+      top: clientRect.top,
+      left: clientRect.left,
+      bottom: clientRect.bottom,
+      right: clientRect.right,
+      width: clientRect.right - clientRect.left,
+      height: clientRect.bottom - clientRect.top,
+    })
+
+    element.setAttribute('data-is-visible', String(isVisible))
+    element.focus = () => ReactTestUtils.Simulate.focus(element)
+  }
+
+  beforeEach(() => {
+    lastFocusedElement = undefined
+  })
+
+  describe('Focusing the ATZ', () => {
+    function setupTest(firstFocusableSelector?: string) {
+      let autoFocusZoneRef: AutoFocusZone | null = null
+      const topLevelDiv = ReactTestUtils.renderIntoDocument(
+        <div onFocusCapture={_onFocus}>
+          <AutoFocusZone
+            data-is-focusable={true}
+            firstFocusableSelector={firstFocusableSelector}
+            ref={ftz => {
+              autoFocusZoneRef = ftz
+            }}
+          >
+            <button className={'f'}>f</button>
+            <FocusZone>
+              <button className={'a'}>a</button>
+              <button className={'b'}>b</button>
+            </FocusZone>
+          </AutoFocusZone>
+          <button className={'z'}>z</button>
+        </div>,
+      ) as HTMLElement
```

Comment:
Can we use enzyme for tests?

Reply:
I was copying and trying to keep these tests consistent with the other focus zone tests.. If we decide later to move all of them to enzyme we will do it for all of them.